indico/modules/events/registration/util.py

```diff
@@ -154,6 +154,7 @@ def get_flat_section_submission_data(regform, *, management=False, registration=
             field_data['availablePlaces'] = item.field_impl.get_available_places(registration)
         else:
             field_data = item.view_data
+        field_data['isLocked'] = registration_data[item.id].is_locked
```

Comment:
I'm pretty sure this will fail in case a field was added after the registration was created. I think the locked flag should be on the field, not the data model (ie `item.is_locked` in here)